include/kernel.h

```diff
@@ -3013,6 +3037,167 @@ extern int k_delayed_work_submit_to_queue(struct k_work_q *work_q,
  */
 extern int k_delayed_work_cancel(struct k_delayed_work *work);
 
+/**
+ * @brief Initialize a delayable work item.
+ *
+ * This routine initializes a delayable work item prior to its first
+ * use.
+ *
+ * @warning This function must not be invoked on a delayable work item that
+ * has been passed to k_work_schedule_for_queue() unless it is known that the
+ * work is no longer delayed, pending on the work queue, or being processed by
+ * the work queue.  This state cannot be inferred from k_work API return
+ * values.
+ *
+ * @param work Address of delayable work item.
+ * @param handler Function to invoke each time work item is processed.
+ *
+ * @return N/A
+ */
+extern void k_work_delayable_init(struct k_work_delayable *work,
+				  k_work_handler_t handler);
+
+/**
+ * @brief Get the parent delayable work structure from a work pointer.
+ *
+ * This function is necessary when a @c k_work_handler_t function is passed to
+ * k_work_schedule_for_queue() and the handler needs to access data associated
+ * with the work item.
+ *
+ * @param work Address passed to the work handler
+ *
+ * @return Address of the containing @c k_delayable_work structure.
```

Comment:
the use of k_delayable_work and k_work_delayable in different places is confusing.

Reply:
Fixed this misspelling.  I'm leaving the name used as an RST anchor in the documentation discussion of delayable work.